internal/store/verticalpodautoscaler.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/kube-state-metrics/pkg/constant"
+
+	autoscaling "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2"
+	vpaclientset "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kube-state-metrics/pkg/metric"
+)
+
+var (
+	descVerticalPodAutoscalerLabelsName          = "kube_verticalpodautoscaler_labels"
+	descVerticalPodAutoscalerLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descVerticalPodAutoscalerLabelsDefaultLabels = []string{"namespace", "verticalpodautoscaler", "target_api_version", "target_kind", "target_name"}
+
+	vpaMetricFamilies = []metric.FamilyGenerator{
+		{
+			Name: descVerticalPodAutoscalerLabelsName,
+			Type: metric.Gauge,
+			Help: descVerticalPodAutoscalerLabelsHelp,
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				labelKeys, labelValues := kubeLabelsToPrometheusLabels(a.Labels)
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   labelKeys,
+							LabelValues: labelValues,
+							Value:       1,
+						},
+					},
+				}
+			}),
+		},
+		{
+			Name: "kube_verticalpodautoscaler_update_mode",
+			Type: metric.Gauge,
+			Help: "Update mode of the VPA.",
```

Comment:
I have a preference for `vertical pod autoscaler` over `vpa` in the metrics help text.

Reply:
Fixed